### DISCUSSION TITLE
admin_config type media

### DISCUSSION THREAD

**Discussion by LaocheXe** [e107inc/e107#5044] [Q&A Plugins]
I have a field that saves as json, for media. However, I don't know why, but I can't save anything in that field in the admin ui - 
the field is like this:
'v_media'                 => array (  'title' => LAN_IMAGE_V,  'type' => 'media', 'tab' => 2, 'data' => 'json',  'width' => 'auto',  'help' => LAN_V_HELP_IMAGES,  'readParms' => 'thumb=80x80',  'writeParms' =>  array ('path' => 'plugin',),  'class' => 'left',  'thclass' => 'left',),

The screen shot below is from e107.org, I don't know if my readParms is listed as thumb=80x80 but i don't know, this is my first attempted on doing this in e107.
![Screenshot 2023-07-16 192209](https://github.com/e107inc/e107/assets/5818022/33c90424-1be2-4542-9026-9ff0230397f7)

In the admin UI it doesn't give me much:
![Screenshot 2023-07-16 191755](https://github.com/e107inc/e107/assets/5818022/7e42c8a8-08aa-41d8-8ecc-6a0da5d94d2f)

In the database field, it just shows [
    {
        "path": ""
    }
]

I don't even know if there is an e107 method to display this like e107::getParser->toImage...
I've uploaded about 4 images, was going to also add a youtube video as well, then have it displayed as thumbnail carousel on a page.

**Comment by tgtje** [comment]
@LaocheXe well _its way out of my league_, but did you read the   e_ajax_class.php , maybe it will give an insight, or idea?
and just a very wild idea (i often think outside the box): are you in the 'allowed to do' range (rights).
Next idea : search in code for    $tp->toImage  
No answers, just ideas...

**Comment by Jimmi08** [comment]
check blank plugin, install it and see if it works. then change things one after the other. 
If not, it is a bug. 

`			'blank_media' 				=> array('title'=> "Media",		        'type' => 'media', 		'data' => 'json',		'width' => '20%',	'thclass' => 'center',	'batch' => TRUE, 'filter'=>TRUE, 'readParms' => array('truncate' => 30), 'writeParms'=>array(/*'w'=>200, 'h'=>150,*/ 'image'=>1, 'video'=>1, 'youtube'=>1, 'audio'=>1, 'glyph'=>1),  'validate' => false),`

I haven't seen the parameter "path"...  only media_category...  and you must it listed in plugin.xml too - owner or something

Sorry, it is a long time... 

Maybe this helps too, just for inspiration:
If I needed custom image folder/field years ago - I had similar questions, no answers - I used this (just tried what worked, no knowledge) :

`'topicimage' => array('title' => LAN_TOPICIMAGE, 'type' => 'method', 'data' => 'str', 'width' => 'auto', 'help' => '', 'readParms' => 'thumb=80x80', 'writeParms' => array(), 'class' => 'left', 'thclass' => 'left'),
`
```
	public function topicimage($curVal, $mode)
	{
		 
		if (!empty($curVal))
		{
			$topic_icon = $curVal;
		}

		switch ($mode)
		{
		case 'read': // List Page

			$path = varset($this->unnuke_prefs['topics_image_path'], '{e_IMAGE}topics/');

			$settings = array('legacyPath' => $path,
				'w' => 200,
				'h' => 50);

			$settings['class'] = 'icon news_image news-category-icon';
			$src = $path . $topic_icon;

			$icon = e107::getParser()->toImage($src, $settings);

			return $icon;
			break;

		case 'write': // Edit Page

			$settings = array('size' => 'xxlarge');
			$maxlength = 80;
			$ret = e107::getForm()->text('topicimage', $topic_icon, $maxlength, $settings);
			return $ret;
		}
	}
```

**Reply by LaocheXe** [reply]
I tired the blank plugin method, I can add whatever, the issue now is that it only allows me to add/select 1 item, instead of multiple.

Also, it saves to the table field like this:

[
    {
        "path": "{e_MEDIA}plugins/dealership/34f038_Screenshot_30.webp"
    }
]

**Reply by tgtje** [reply]
@LaocheXe just informational. what would happen if you use _ftp upload_   (multiple files) to the upload folder (system/hash/**import**) and then use upload/import function from within media manager??(downfall : if t works, ok for a 'site owner' , not a good thing for a lot of others of course).
As for path.. just by reading i would assume the image resides in the plugin folder (or stored there).

**Reply by LaocheXe** [reply]
When I first started this project, I was thinking of just doing that, or just make gallery categories, and then just select them in the plugin area. but I thought it would be best not to have to go back-and-forth just to add just images.

I want to make a thumbnail carousel like this: https://store.steampowered.com/app/1286680/Tiny_Tinas_Wonderlands/
To display youtube video, and images, and going through the admin ui documents, it look like media with json was the best. I would just go, cut it out, and used the gallery category method, but the gallery doesn't support youtube videos, or does it?

**Reply by tgtje** [reply]
well the carousel displayed is a video gallery system, and helas, you will have to go back and forth, didn't try it or know it will work, else you also could use maybe featurebox for it.

Or do not re-invent and look external and bind in ; example > unite gallery

**Reply by LaocheXe** [reply]
I want to get the admin area to function, can only save one file, I can upload multiple, but can only select one for the 'type' => 'media', 'data' => 'json'. I want to have multiple, unless that is not it's intended function.

**Reply by tgtje** [reply]
As mentioned, not my thing, but having a little read in about json, it looks for me as json is like a container, so yes could be possible it is a 1 time action (it has to do something somewhere), and multiple different pieces (or calls) may not work alongside. 
That seems you would need to work inside that part using arrays (or objects). If i understand correctly each image uses an own url and those could be placed in an array, resulting in 1 json type file, holding multiple urls for images and so on.. 
I is just what i extract from those readings, and hope i understand its function/s. I could be totally wrong and hope anoher one can shed some more information, being more used to do things in another way or by system.
